File: sdks/node-sdk/src/logger.ts
export interface AgentLoggerConfig {
	name: string;
}

export const LogLevels = [
	'error',
	'warn',
	'info',
	'debug',
] as const;

export type LogLevel = typeof LogLevels[number];

export class Logger {
	constructor(private readonly config: AgentLoggerConfig) {
	}

	private generateTimestamp() {
		const now = new Date();
		const date = `${now.getMonth() + 1}/${now.getDate()}/${now.getFullYear()}`;
		const time = `${now.getHours()}:${now.getMinutes()}:${now.getSeconds()}`
		const timestamp = `${date} ${time}`;

		return timestamp;
	}

	private log(level: LogLevel, ...messages: string[]) {
		const ts = this.generateTimestamp()
		const message = messages.join(' ');
		const logFn = console[level] || console.log;

		logFn(`[${this.config.name}] [${ts}] ${level} ${message}`);
	}

	error(...messages: string[]) {
		this.log('error', ...messages);
	}

	warn(...messages: string[]) {
		this.log('warn', ...messages);
	}

	info(...messages: string[]) {
		this.log('info', ...messages);
	}

	debug(...messages: string[]) {
		this.log('debug', ...messages);
	}
}

File: sdks/python-sdk/agentlabs/_internals/logger.py
from datetime import datetime

class Logger:
    def __init__(self, name: str) -> None:
        self.name = name

    def _format_log(self, level: str, message: str) -> str:
        return f"[{self.name}] {datetime.now().strftime('%m/%d/%Y, %H:%M:%S')} {level} {message}"

    def _log(self, level: str, message: str) -> None:
        print(self._format_log(level, message))

    def info(self, message: str) -> None:
        self._log("INFO", message)

    def warn(self, message: str) -> None:
        self._log("WARN", message)

    def error(self, message: str) -> None:
        self._log("ERROR", message)

    def debug(self, message: str) -> None:
        self._log("DEBUG", message)
